CardParts/src/Classes/Card Parts/CardPartMapView.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  CardPartMapView.swift
+//  CardParts
+//
+//  Created by Michael VanDyke on 10/5/19.
+//
+
+import Foundation
+import MapKit
+import RxCocoa
+import RxSwift
+
+public class CardPartMapView: UIView, CardPartView {
+    public var mapType: MKMapType {
+        didSet {
+            updateMap()
+        }
+    }
+    
+    public var location: CLLocation {
+        didSet {
+            updateMap()
+        }
+    }
+    
+    public var span: MKCoordinateSpan = MKCoordinateSpan(latitudeDelta: 1, longitudeDelta: 1) {
+        didSet {
+            updateMap()
+        }
+    }
+    
+    public var margins: UIEdgeInsets = CardParts.theme.cardPartMargins
+    
+    public var mapView: MKMapView
+    
+    public init(type: MKMapType, location: CLLocation, span: MKCoordinateSpan?) {
+        mapView = MKMapView(frame: .zero)
+        self.mapType = type
+        self.location = location
+        
+        if let span = span {
+            self.span = span
+        }
+        
+        super.init(frame: .zero)
+        
+        addSubview(mapView)
+        setNeedsUpdateConstraints()
+        updateMap()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public override func updateConstraints() {
+        mapView.layout {
+            $0.top == self.topAnchor
+            $0.leading == self.leadingAnchor
+            $0.trailing == self.trailingAnchor
+            $0.bottom == self.bottomAnchor
+        }
```

Comment:
Awesome , looks very clean.

Reply:
Yeah, that was very nice cleanup. Thanks for that.